internal/controller/controllers/hypershiftagentserviceconfig_controller_test.go

```diff
@@ -51,31 +57,85 @@ var _ = Describe("HypershiftAgentServiceConfig reconcile", func() {
 				},
 
 				KubeconfigSecretRef: corev1.LocalObjectReference{
-					Name: testKubeconfigSecret,
+					Name: testKubeconfigSecretName,
 				},
 			},
 		}
 	}
 
+	newKubeconfigSecret := func() *corev1.Secret {
+		return &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      testKubeconfigSecretName,
+				Namespace: testNamespace,
+			},
+			Data: map[string][]byte{
+				"kubeconfig": []byte(BASIC_KUBECONFIG),
+			},
+			Type: corev1.SecretTypeOpaque,
+		}
+	}
+
+	newHSCTestReconciler := func(mockSpokeClientCache *MockSpokeClientCache, initObjs ...runtime.Object) *HypershiftAgentServiceConfigReconciler {
+		schemes := GetKubeClientSchemes()
+		c := fakeclient.NewClientBuilder().WithScheme(schemes).WithRuntimeObjects(initObjs...).Build()
+		return &HypershiftAgentServiceConfigReconciler{
+			Client:       c,
+			Scheme:       c.Scheme(),
+			Log:          common.GetTestLog(),
+			SpokeClients: mockSpokeClientCache,
+		}
+	}
+
 	BeforeEach(func() {
-		c = fakeclient.NewClientBuilder().WithScheme(GetKubeClientSchemes()).Build()
 		mockCtrl = gomock.NewController(GinkgoT())
+		mockSpokeClient = spoke_k8s_client.NewMockSpokeK8sClient(mockCtrl)
```

Comment:
I know you're not using it for anything currently, but you may eventually want to make this a fakeclient for better tests later.